Geometry_Engine/Query/LineIntersection.cs

```diff
@@ -258,6 +259,160 @@ public static List<Point> LineIntersections(this Circle circle, Line line, bool
 
         /***************************************************/
 
+        [Description("Calculates and returns the intersection points of a Line and a Ellipse.")]
+        [Input("ellipse", "Ellipse to intersect with the Line.")]
+        [Input("line", "Line to intersect with the Ellipse.")]
+        [Input("useInfiniteLine", "If true or if the Infinite property of the Line is true, a intersection point found that is outside the domain of the Lines start and end point is accepted. If false, and the Lines Infinite property is false, the found intersection point needs to be on the finite line segment, between or on the start and end point.")]
+        [Input("tolerance", "Distance tolerance to be used in the method. Used for checking if the intersection point is within acceptable distance from the Lines.", typeof(Length))]
+        [Output("intersections", "The intersection points of the Ellipse and the Line.")]
+        public static List<Point> LineIntersections(this Ellipse ellipse, Line line, bool useInfiniteLine = false, double tolerance = Tolerance.Distance)
+        {
+            if (ellipse.IsNull() || line.IsNull())
+                return new List<Point>();
```

Comment:
I would return `null` output for `null` inputs

Reply:
As previous, could argue both ways. Lets grab a quick chat when you are in :)